runtime/tr.source/trj9/optimizer/J9TransformUtil.cpp

```diff
@@ -1077,6 +1099,42 @@ J9::TransformUtil::transformIndirectLoadChainImpl(TR::Compilation *comp, TR::Nod
       }
 
    TR::SymbolReference *symRef = node->getSymbolReference();
+   // Add known object constraint to element from handleTable or typeTable
+   // Opt can be extended to elements of any arrays that are immutable
+   if (!symRef->isUnresolved() &&
+       node->getSymbol()->isArrayShadowSymbol() &&
+       node->getFirstChild()->getOpCode().isArrayRef() &&
+       node->getFirstChild()->getSecondChild()->getOpCode().isLoadConst() &&
+       (baseExpression->getSymbol()->getRecognizedField() == TR::Symbol::Java_lang_invoke_VarHandle_handleTable ||
+        baseExpression->getSymbol()->getRecognizedField() == TR::Symbol::Java_lang_invoke_VarHandle_typeTable))
+       {
+       uintptrj_t offset = 0;
+       TR::Node* offsetNode = node->getFirstChild()->getSecondChild();
+       if (offsetNode->getDataType() == TR::Int64)
+          offset = offsetNode->getUnsignedLongInt();
+       else
+          offset = offsetNode->getUnsignedInt();
+
+       uintptrj_t value = fej9->getReferenceFieldAtAddress(baseAddress + offset);
+       if (value)
+          {
+          TR::SymbolReference *improvedSymRef = comp->getSymRefTab()->findOrCreateSymRefWithKnownObject(node->getSymbolReference(), &value);
```

Comment:
`node->getSymbolReference()` has already been fetched into local `symRef`.  It's easier to reason about the code , at least for a non-jit dev,  if the locals get reused rather than fetching them again as that way its clearly the same symref.